test/local/routes/recovery_email_status.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+require('ass')
+
+var sinon = require('sinon')
+
+var test = require('../../ptaptest')
+var mocks = require('../../mocks')
+var config = require('../../../config').getProperties()
+
+var P = require('../../../lib/promise')
+var error = require('../../../lib/error')
+
+function getMockedRoute(path, mockDB) {
+  var routes = require('../../../lib/routes/account')(
+    mocks.mockLog(),
+    require('crypto'),
+    P,
+    undefined,
+    require('joi'),
+    require('../../../lib/error'),
+    mockDB,
+    undefined,
+    undefined,
+    config,
+    undefined,
+    undefined,
+    undefined
+  )
+  for (var i = 0; i < routes.length; i++) {
+    if (routes[i].path === path) {
+      return routes[i]
+    }
+  }
+  throw new Error('Could not find route: ' + path)
+}
```

Comment:
@rfk can we do something like this here: (we can move `require` into `makeRoutes`)

```

var P = require('../../lib/promise')
var uuid = require('uuid')
var crypto = require('crypto')
var isA = require('joi')
var error = require('../../lib/error')

var TEST_EMAIL = 'foo@gmail.com'

var makeRoutes = function (options) {
  options = options || {}

  var config = options.config || {
    smtp: {}
  }
  var log = options.log || {}
  var Password = require('../../lib/crypto/password')(log, config)
  return require('../../lib/routes/account')(
    log,
    crypto,
    P,
    uuid,
    isA,
    error,
    options.db || {},
    options.mailer || {},
    Password,
    config
  )
}

var getRoute = function (routes, path) {
  var route = null

  routes.some(function (r) {
    if (r.path === path) {
       route = r
       return true;
    }
  })

  return route;
}
```
Usage:
```

    var recoveryEmail = getRoute(accountRoutes, '/recovery_email/status')

    var mockRequest = {}
    var mockResponse = {}

    recoveryEmail.handler(mockRequest, mockResponse)
```

Thoughts?